pkg/git/git.go

```diff
@@ -77,6 +79,28 @@ func (g *GitClient) DiffDirsAbs(ctx context.Context, sourceRef, destRef string)
 	return parseSortedDiffDirsAbs(ctx, stdout.String())
 }
 
+// CloneRepository clones the repository to the workingDir.
+func (g *GitClient) CloneRepository(ctx context.Context, githubToken, owner, repo string) error {
+	logger := logging.FromContext(ctx).With("working_dir", g.workingDir)
+
+	var stdout, stderr bytes.Buffer
+
+	_, err := child.Run(ctx, &child.RunConfig{
+		Stdout:     &stdout,
+		Stderr:     &stderr,
+		WorkingDir: g.workingDir,
+		Command:    "git",
+		Args:       []string{"clone", fmt.Sprintf("https://%s@github.com/%s/%s.git", githubToken, owner, repo)},
```

Comment:
Should we use `--single-branch` and `--no-tags` just for efficiency. We dont need everything just everything from the default branch right?

Reply:
Done